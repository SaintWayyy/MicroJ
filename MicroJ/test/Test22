/* This test test the polymorphism  */
class Coordinate{
    int x;
    int y;
    constructor Coordinate(int x, int y){}
    void printLocation(Coordinate self){
        print("X-cordinate:");
        print(self.x);
        print("Y-cordinate:");
        print(self.y);
    }
}

interface operation{
    void moveX(int distance);
    void moveY(int distance);
}

class Point extends Coordinate implements operation{
    constructor Point(int x, int y) {}

    public void moveX(Point self, int distance) {
        self.x = self.x + distance;
    }

    public void moveY(Point self, int distance) {
        self.y = self.y + distance;
    }

    public void printLocation(Point self){
        print("Point Location:");
    }
}

int main(){
    Coordinate zero : = new Point(0, 0);
    Coordinate orign : = new Coordinate(0,0);
    zero.moveX(3);
    
    zero.printLocation();
    /* The out put shoud be "Point Location:" */
    orign.printLocation();
    /* The out put should be:
    "X-cordinate:"
    0
    "Y-cordinate:"
    0
    */
}