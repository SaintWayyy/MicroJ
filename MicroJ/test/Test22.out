"Point Location:"
"X-cordinate:"
0
"Y-cordinate:"
0
